### PR TITLE
Adding the ability to store additional data in the session cookie.

### PR DESCRIPTION
**Pull request by lukebussey**
This pull adds the ability to store additional data upon initialization into the session cookie. E.g.

```
window.lastCampaign({
    domain: YW.cookieDomain,
    params: [
        'gclid',
        'campaign',
        'source',
        'medium',
        'term'
    ],
    data: {
        site_section: document.body.getAttribute('data-site-section') || null
    }
});
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -59,30 +55,45 @@ function saveLastCampaign(opts) {
...
+                }
+
+                setCookie(cookie.serialize(options.prefix + key, data[key], cookieOptions));
             }
+        });
+    }
 
-            setCookie(cookie.serialize(options.prefix + key, data[key], cookieOptions));
+    // Save the data object
+    if (Object.keys(options.data).length !== 0) {
```
Comment (mattduggan) [inline index.js]: [nit] `for ... in` will iterate over prototype chain.

You could reuse the array returned from Object.keys().

```
var dataKeys = Object.keys(options.data);
if (dataKeys.length == 0) {
    return;
}
dataKeys.forEach()
```

**Comment by mattduggan** [comment]
One comment

**Comment by lukebussey** [comment]
Fixed it.

**Comment by mattduggan** [comment]
g2g

**Pull request merged**